### PR TITLE
[Merged by Bors] - feat: port Topology.Homotopy.HSpaces

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
Should we wait for leanprover-community/mathlib#19075?

**Comment by urkud** [comment]
Do you need help with rebasing on top of the current `mathport` output?

**Comment by mo271** [comment]
> Do you need help with rebasing on top of the current `mathport` output?

done

File: Mathlib/Topology/Homotopy/HSpaces.lean [hunk 1]
```diff
@@ -65,7 +68,7 @@ instance HSpace.prod (X : Type u) (Y : Type v) [TopologicalSpace X] [Topological
             (continuous_fst.prod_mk (continuous_fst.comp continuous_snd))).prod_mk
         (Continuous.comp HSpace.eHmul.1.1.2
           (continuous_fst.prod_mk (continuous_snd.comp continuous_snd)))
-    use ⟨G, hG⟩
+    use ⟨G, hG, _⟩
```
Comment (urkud) [inline Mathlib/Topology/Homotopy/HSpaces.lean]: Does the following work? Same below.
```suggestion
    use [G, hG]
```

File: Mathlib/Topology/Homotopy/HSpaces.lean [hunk 2]
```diff
@@ -84,7 +87,7 @@ instance HSpace.prod (X : Type u) (Y : Type v) [TopologicalSpace X] [Topological
             (continuous_fst.prod_mk (continuous_fst.comp continuous_snd))).prod_mk
         (Continuous.comp HSpace.hmulE.1.1.2
           (continuous_fst.prod_mk (continuous_snd.comp continuous_snd)))
-    use ⟨G, hG⟩
+    use ⟨G, hG, _⟩
```
Comment (urkud) [inline Mathlib/Topology/Homotopy/HSpaces.lean]:
```suggestion
    use [G, hG]
```

**Review by urkud — COMMENTED** [review]
I'm sorry. It seems that https://github.com/leanprover-community/mathlib/pull/19075 didn't go through `mathport` yet because we have some issues with CI. Do you prefer to merge this PR, then forward-port 19075 or to wait&rebase?

**Comment by mo271** [comment]
> I'm sorry. It seems that [leanprover-community/mathlib#19075](https://github.com/leanprover-community/mathlib/pull/19075) didn't go through `mathport` yet because we have some issues with CI. Do you prefer to merge this PR, then forward-port 19075 or to wait&rebase?

I'd advocate for wait&rebase

File: Mathlib/Topology/Homotopy/HSpaces.lean (hunk 2 above)
Comment (mo271) [reply Mathlib/Topology/Homotopy/HSpaces.lean]: no, as above

**Comment by mo271** [comment]
@urkud : meanwhile mathport has picked up the change and I rebased..

File: Mathlib/Topology/Homotopy/HSpaces.lean [hunk 3]
```diff
@@ -244,16 +276,16 @@ theorem continuous_delayReflLeft : Continuous fun p : I × Path x y => delayRefl
...
 theorem delayReflLeft_one (γ : Path x y) : delayReflLeft 1 γ = γ := by
-  simp only [delay_refl_left, delay_refl_right_one, Path.symm_symm]
+  simp only [delayReflLeft, delayReflRight_one, Path.symm_symm]
 #align path.delay_refl_left_one Path.delayReflLeft_one
 
-/-- The loop space at x carries a structure of a `H-space`. Note that the field `e_Hmul`
-(resp. `Hmul_e`) neither implies nor is implied by `path.homotopy.refl_trans`
-(resp. `path.homotopy.trans_refl`).
+/-- The loop space at x carries a structure of a `H-space`. Note that the field `eHmul`
+(resp. `HmulE`) neither implies nor is implied by `Path.Homotopy.reflTrans`
```
Comment (semorrison) [inline Mathlib/Topology/Homotopy/HSpaces.lean:290]:
```suggestion
(resp. `hmulE`) neither implies nor is implied by `Path.Homotopy.reflTrans`
```

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**